app/src/main/kotlin/com/wire/android/ui/userprofile/image/AvatarPicker.kt

```diff
@@ -58,8 +57,8 @@ fun AvatarPickerScreen(viewModel: AvatarPickerViewModel) {
             viewModel.navigateBack()
         },
         onSaveClick = {
-            if (state.avatarPickerFlow.pictureState is PictureState.Picked) {
-                viewModel.uploadNewPickedAvatarAndBack(state.avatarPickerFlow.pictureState.avatarUri, context)
+            if (canUpload) {
+                viewModel.uploadNewPickedAvatarAndBack(viewModel.pictureState.avatarUri)
```

Comment:
from what I read from the function, the canUpload check is not needed as triggering this lambda assumes already that the state of the PictureState is equal to Picked

```
@Composable
private fun AvatarPickerActionButtons(
    hasPickedImage: Boolean,
    onSaveClick: () -> Unit,
    onCloseClick: () -> Unit,
    onChangeImage: () -> Unit
) {
    if (hasPickedImage) {
        Row(Modifier.fillMaxWidth()) {
            WireSecondaryButton(
                modifier = Modifier
                    .padding(dimensions().spacing16x)
                    .weight(1f),
                text = stringResource(R.string.label_cancel),
                onClick = { onCloseClick() }
            )
            WirePrimaryButton(
                modifier = Modifier
                    .padding(dimensions().spacing16x)
                    .weight(1f),
                text = stringResource(R.string.label_confirm),
                onClick = { onSaveClick() }
            )
        }
    } else {
        WirePrimaryButton(
            modifier = Modifier.padding(dimensions().spacing16x),
            text = stringResource(R.string.profile_image_change_image_button_label),
            onClick = { onChangeImage() }
        )
    }
}
```